js/coinfloor.js

```diff
@@ -201,6 +201,137 @@ module.exports = class coinfloor extends Exchange {
         return this.parseTrades (response, market, since, limit);
     }
 
+    async fetchLedger (code = undefined, since = undefined, limit = undefined, params = {}) {
+        // code is actually a market symbol in this situation, not a currency code
+        await this.loadMarkets ();
+        let market = undefined;
+        if (code) {
+            market = this.findMarket (code);
+            if (!market) {
+                throw new NotSupported (this.id + ' fetchTransactions requires a market symbol param');
+            }
+        }
+        const request = {
+            'id': market['id'],
+            'limit': limit,
+        };
+        const response = await this.privatePostIdUserTransactions (this.extend (request, params));
+        return this.parseLedger (response, undefined, since, limit);
+    }
+
+    parseLedgerEntryStatus (status) {
+        const types = {
+            'completed': 'ok',
+        };
+        return this.safeString (types, status, status);
+    }
+
+    parseLedgerEntryType (type) {
+        const types = {
+            '0': 'transaction', // deposit
+            '1': 'transaction', // withdrawal
+            '2': 'trade',
+        };
+        return this.safeString (types, type, type);
+    }
+
+    parseLedgerEntry (item, currency = undefined) {
+        // trade:
+        //   {
+        //     "datetime": "2017-07-25 06:41:24",
+        //     "id": 1500964884381265,
+        //     "type": 2,
+        //     "xbt": "0.1000",
+        //     "xbt_eur": "2322.00",
+        //     "eur": "-232.20",
+        //     "fee": "0.00",
+        //     "order_id": 84696745 }
+        //
+        // transactions:
+        //   {
+        //     "datetime": "2017-07-25 13:19:46",
+        //     "id": 97669,
+        //     "type": 1,
+        //     "xbt": "-3.0000",
+        //     "xbt_eur": null,
+        //     "eur": "0",
+        //     "fee": "0.0000",
+        //     "order_id": null }
+        //   {
+        //     "datetime": "2017-07-27 16:44:55",
+        //     "id": 98277,
+        //     "type": 0,
+        //     "xbt": "0",
+        //     "xbt_eur": null,
+        //     "eur": "4970.04",
+        //     "fee": "0.00",
+        //     "order_id": null }
+        const keys = Object.keys (item);
+        let baseId = undefined;
+        let quoteId = undefined;
+        for (let i = 0; i < keys.length; i++) {
+            const key = keys[i];
+            if (key.indexOf ('_') > 0) {
+                const parts = key.split ('_');
+                const numParts = parts.length;
```

Comment:
We need to do this hinting, as explained here: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#working-with-array-lengths